ArcadiaParties.CQRS/Handlers/GetCurrentUserHandler.cs

```diff
@@ -1,26 +1,46 @@
-﻿using ArcadiaParties.CQRS.Queries;
-using ArcadiaParties.Data.Abstractions.DTOs;
-using ArcadiaParties.Data.Abstractions.Repositories;
-using MediatR;
-using System.Collections.Generic;
+using System;
 using System.Threading;
 using System.Threading.Tasks;
+using ArcadiaParties.CQRS.Queries;
+using ArcadiaParties.Data.Abstractions.DTOs;
+using AutoMapper;
+using MediatR;
 
 namespace ArcadiaParties.CQRS.Handlers
 {
-    internal class GetCurrentUserHandler : IRequestHandler<GetCurrentUserQuery, IEnumerable<string>>
+    internal class GetCurrentUserHandler : IRequestHandler<GetCurrentUserQuery, UserDTO>
     {
-        private readonly IUserRepository _repo;
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
 
-        public GetCurrentUserHandler(IUserRepository repo)
+        public GetCurrentUserHandler(IMediator mediator, IMapper mapper)
         {
-            _repo = repo;
-        }
+            _mediator = mediator;
+            _mapper = mapper;
 
-        public async Task<IEnumerable<string>> Handle(GetCurrentUserQuery request, CancellationToken cancellationToken)
+        }
+        public async Task<UserDTO> Handle(GetCurrentUserQuery request, CancellationToken cancellationToken)
         {
-            var user = await _repo.GetUser(request.Principal.Identity.Name);
-            return user.Roles;
+            var userQuery = new GetUserRolesQuery(request.User);
```

Comment:
Why? Why is it named `userQuery` if it gets user **roles**?

Reply:
Renamed